src/mesh/analysis.rs

```diff
@@ -1,13 +1,125 @@
-use std::collections::HashSet;
+use std::collections::{HashMap, HashSet};
 
-use nalgebra::base::Vector3;
-use nalgebra::geometry::Point3;
+use nalgebra as na;
+use nalgebra::{Point3, Vector3};
 
 use crate::convert::{cast_i32, cast_usize};
-use crate::edge_analysis::EdgeSharingMap;
-use crate::geometry::{Face, Geometry, OrientedEdge, UnorientedEdge};
 
-/// Finds edges with a certain valency in a mesh edge collection
+use super::{Face, Mesh, OrientedEdge, UnorientedEdge};
+
+pub fn compute_bounding_sphere<'a, I>(meshes: I) -> (Point3<f32>, f32)
+where
+    I: IntoIterator<Item = &'a Mesh> + Clone,
+{
+    let centroid = compute_centroid(meshes.clone());
+    let mut max_distance_squared = 0.0;
+
+    for mesh in meshes {
+        for vertex in &mesh.vertices {
+            let distance_squared = na::distance_squared(&centroid, vertex);
+            if distance_squared > max_distance_squared {
+                max_distance_squared = distance_squared;
+            }
+        }
+    }
+
+    (centroid, max_distance_squared.sqrt())
+}
+
+pub fn compute_centroid<'a, I>(meshes: I) -> Point3<f32>
+where
+    I: IntoIterator<Item = &'a Mesh>,
+{
+    let mut vertex_count = 0;
+    let mut centroid = Point3::origin();
+    for mesh in meshes {
+        vertex_count += mesh.vertices.len();
+        for vertex in &mesh.vertices {
+            let v = vertex - Point3::origin();
+            centroid += v;
+        }
+    }
+
+    centroid / (vertex_count as f32)
+}
+
+// FIXME: deprecate and remove, superseded by pull point to mesh
```

Comment:
_I know it's not related to this PR but it's just a matter of changing this FIXME note:_ I wouldn't remove this, rather just rename and change the arguments to `point: Point3, cloud: [Point3]`. I believe it still might be useful.